provisioning/device/samples/readme.md

```diff
@@ -0,0 +1,101 @@
+`<"My CA Certificate", "My Intermediate 1", ..., "My Intermediate N", "MyDevice1>`.
+
+2. Individual Enrollment
+In this case, each device certificate gets its own certificate uploaded to the service.
+_Note:_ the device certificate can also be a signing certificate of the actual certificate used to authenticate with the service during provisioning.
+
+An example of specifying the authentication X509Certificate using a PKCS12 PFX password-protected file:
+
+```C#
+    using (var certificate = new X509Certificate2(s_certificateFileName, certificatePassword))
+    using (var security = new SecurityClientX509(certificate))
+    {
+        //...
+    }
+```
+
+The SDK provides an extension model [SecurityClientHsmX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientHsmX509.cs) that allows hardware vendors to implement custom Hardware Security Modules that store the device certificates. On Windows, PKCS11 HSM devices are supported through the [Certificate Store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores).
+
+An example of implementation for this extension module is the [SecurityClientX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientX509.cs) class.
+
+### Provisioning devices using TPM based attestation
+
+In the case of TPM attestation, a RegistrationID must be supplied by the application.
+
+The TPM attestation supports only Individual Enrollments. The [Endorsement Key](https://technet.microsoft.com/en-us/library/cc770443(v=ws.11).aspx) (EK) must be supplied to the service. During provisioning, the [Storage Root Key](https://technet.microsoft.com/en-us/library/cc753560(v=ws.11).aspx) (SRK) will also be used to ensure that TPM ownership changes result in devices provisioned to the correct owner.
+
+
+```C#
+    using (var security = new SecurityClientTpmSimulator(RegistrationId))
+    {
+        //...
+    }
+```
+
+The SDK provides an extension model [SecurityClientHsmTpm](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientHsmTpm.cs) that allows hardware vendors to implement custom TPM v2.0 Hardware Security Modules.
+
+The samples use a TPMv2.0 simulator that uses a loopback TCP connection for communication. This is provided for demonstration purposes only and does not provide any security.
+
+## List of samples
+
+- [Provisioning Devices with X509 certificate attestation](ProvisioningDeviceClientX509)
+- [Provisioning Devices with TPM attestation](ProvisioningDeviceClientTpm)
+
+### How to run the samples
+
+_Preview only:_ Running the samples requires building from sources. This will not be necessary once NuGet packages are released.
+
+1. Prepare your development environment. Follow the instructions at https://github.com/Azure/azure-iot-sdk-csharp/blob/master/device/doc/devbox_setup.md
```

Comment:
What does a bug basher need in order to get started quickly?  Can we provide a list of recommended steps (build locally v. get package, if get package, then how do we proceed etc.)?

Reply:
At the moment we don't have packaging and so, unfortunately, this is the shortest path - as mentioned in line 85.

At the very minimum, the system should have the following on Windows or Linux:
1. http://dot.net
2. Powershell (https://github.com/PowerShell/PowerShell)

.Net Framework / Xamarin will not be testable with only that.

Reply:
Do we need to build the SDK in this step?

Reply:
Yes, everything needs to be built at the moment. We won't need to do this after we have packages.